### PR TITLE
[CL:Test] Add sqrtPrice check to swaps_test

### PR DESCRIPTION
**Pull request by stackman27**
[Closes: #3989](https://github.com/osmosis-labs/osmosis/issues/3989)

## What is the purpose of the change
checks sqrtPrice for TestCalcAndSwapOutAmtGivenIn, TestCalcAndSwapInAmtGivenOut

## Brief Changelog
n/a

## Testing and Verifying
n/a

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by czarcas7ic** [comment]
Are we able to tie in with this PR these exact values but also derived from the python script Roman created? Otherwise we are just taking the word of whatever what output as the result which could potentially be incorrect.

**Comment by mattverse** [comment]
Good call!

> Are we able to tie in with this PR these exact values but also derived from the python script Roman created? Otherwise we are just taking the word of whatever what output as the result which could potentially be incorrect.

Good call! Will review once @czarcas7ic 's points has been addressed

File: scripts/cl/one_for_zero.py [hunk 1]
```diff
@@ -2,30 +2,36 @@
 import sympy as sp
 from common import *
 
+
```
Comment (stackman27) [inline scripts/cl/one_for_zero.py:5]: note: all the python changes are from my prettifier. lmk if there's a specific preference and i'll reset them

**Comment by stackman27** [comment]
Added wolfram calculation. Ready for rereview @czarcas7ic @mattverse!

**Comment by stackman27** [comment]
Added wolfram calculation. Ready for rereview @czarcas7ic @mattverse!

**Comment by p0mvn** [comment]
Please do not merge this before the swap issues we're working on with Adam are resolved to avoid test breakage for the large PRs with fixes

**Comment by stackman27** [comment]
> Please do not merge this before the swap issues we're working on with Adam are resolved to avoid test breakage for the large PRs with fixes

Please lmk when the other PRs gets merged so that i can rebase and get this one ready!

**Comment by p0mvn** [comment]
This is ready to be merged once conflicts are resolved. 

I attempted rebasing but most tests ended up failing. I'm not going to push to avoid breaking something here

File: x/concentrated-liquidity/swaps_test.go [hunk 2]
```diff
@@ -71,9 +71,10 @@ var (
 			// sqrtPriceCurrent: 70.710678118654752440 which is 5000
 			// expectedTokenIn:  41999999.9999 rounded up https://www.wolframalpha.com/input?i=1517882343.751510418088349649+*+%2870.738349405152439867+-+70.710678118654752440%29
 			// expectedTokenOut: 8396.71424216 rounded down https://www.wolframalpha.com/input?i=%281517882343.751510418088349649+*+%2870.738348247484497717+-+70.710678118654752440+%29%29+%2F+%2870.710678118654752440+*+70.738348247484497717%29
-			expectedTokenIn:  sdk.NewCoin("usdc", sdk.NewInt(42000000)),
-			expectedTokenOut: sdk.NewCoin("eth", sdk.NewInt(8396)),
-			expectedTick:     sdk.NewInt(310040),
+			expectedTokenIn:   sdk.NewCoin("usdc", sdk.NewInt(42000000)),
+			expectedTokenOut:  sdk.NewCoin("eth", sdk.NewInt(8396)),
+			expectedTick:      sdk.NewInt(310040),
+			expectedSqrtPrice: sdk.MustNewDecFromStr("70.738348247484497717"), // https://www.wolframalpha.com/input?i=70.710678118654752440+%2B++++%2842000000+%2F+1519437308.014768571721000000%29
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps_test.go:77]: This link does not match the result

File: x/concentrated-liquidity/swaps_test.go [hunk 3]
```diff
@@ -174,6 +178,7 @@ var (
 			expectedTokenIn:            sdk.NewCoin("usdc", sdk.NewInt(10000000000)),
 			expectedTokenOut:           sdk.NewCoin("eth", sdk.NewInt(1820630)),
 			expectedTick:               sdk.NewInt(321055),
+			expectedSqrtPrice:          sdk.MustNewDecFromStr("78.137149196095607129"),
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps_test.go:181]: How is this computed?

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/swaps_test.go (hunk 3 above)
Comment (stackman27) [reply x/concentrated-liquidity/swaps_test.go:181]: hrmm this value seems to be a lil off from what i generate in python. Could you please verify? 

https://www.wolframalpha.com/input?i=74.161984870956629487+%2B+%284761322417.810613244228191057067224+%2F+1197767444.955508123222985080%29

Comment (stackman27) [reply x/concentrated-liquidity/swaps_test.go:181]: i'm following these params: 
liquidity: 1197767444.955508123222985080
sqrtPriceCurrent: 74.161984870956629487 (5500) 
amtRemaning: 4761322417.810613244228191057067224

Comment (stackman27) [reply x/concentrated-liquidity/swaps_test.go]: just fixed!

File: x/concentrated-liquidity/swaps_test.go (hunk 2 above)
Comment (stackman27) [reply x/concentrated-liquidity/swaps_test.go]: fixed

**Comment by stackman27** [comment]
@p0mvn resolved your comments. One extra thing to point out is the test comments are very misleading, we should definitely think about either removing them or adding new ones

**Comment by p0mvn** [comment]
> @p0mvn resolved your comments. One extra thing to point out is the test comments are very misleading, we should definitely think about either removing them or adding new ones

Agreed, that's why for fee tests I've been moving towards estimating things via Python.

As we keep reviewing and refactoring swaps, I think we might want to do something similar for all other tests

File: x/concentrated-liquidity/swaps_test.go [hunk 4]
```diff
@@ -781,12 +814,13 @@ var (
...
 			expectedTokenIn:            sdk.NewCoin("usdc", sdk.NewInt(64417624)),
 			expectedTick:               sdk.NewInt(309941),
+			expectedSqrtPrice:          sdk.MustNewDecFromStr("70.668238976219012614"), // https://www.wolframalpha.com/input?i=%28%281517882343.751510418088349649%29%29+%2F+%28%28%281517882343.751510418088349649%29+%2F+%2870.710678118654752440%29%29+%2B+%2812891.26207649936510%29%29
 			expectedLowerTickFeeGrowth: DefaultFeeAccumCoins,
 			expectedUpperTickFeeGrowth: DefaultFeeAccumCoins,
 		},
 	}
 
-	swapInGivenOutFeeTestCases = map[string]SwapTest{
+
```
Comment (stackman27) [inline x/concentrated-liquidity/swaps_test.go:823]:
```suggestion
	swapInGivenOutFeeTestCases = map[string]SwapTest{
```

File: x/concentrated-liquidity/swaps_test.go [hunk 5]
```diff
@@ -1002,7 +1051,7 @@ func (s *KeeperTestSuite) TestCalcAndSwapOutAmtGivenIn() {
 
 			// perform swap
 			// TODO: Add sqrtPrice check
```
Comment (stackman27) [inline x/concentrated-liquidity/swaps_test.go:1053]:
```suggestion
```

**Review by p0mvn — APPROVED** [review]
LGTM. Thank you @stackman27 and sorry for breaking your branch earlier

**Pull request merged**